spec/cspec/c/gen-config-thy.py

```diff
@@ -235,25 +253,27 @@ def add_defaults(config: Dict[str, str]):
   GENERATED -- DO NOT EDIT! Changes will be overwritten.
 
   This file was generated from {}
+  and {}
   by the script {}.
 *)
 
 """
 
 
-def write_config_thy(config_thy_path, config: Dict[str, str]):
+def write_config_thy(header, config_thy_path, config: Dict[str, str], physBase=None):
     """
     Write a Kernel_Config.thy file for a given configuration dict.
     """
     file_name = path.realpath(path.join(config_thy_path, 'Kernel_Config.thy'))
     with open(file_name, 'w') as f:
-        f.write(theory_header.format(
-            date.today().year,
-            config_path,
-            path.realpath(__file__))
-        )
+        f.write(header)
+
+        if physBase:
+            f.write('(* This value is PHYS_BASE_RAW in the devices_gen.h header listed above. *)\n')
+            f.write('definition physBase :: machine_word where\n')
+            f.write(f'  "physBase \\<equiv> {physBase}"\n\n')
 
-        names = []
+        names = ['physBase'] if physBase else []
```

Comment:
too fast! I typed this very line, and when I tried to add a comment is said I should refresh first. Nice!